extensions/stackdriver/stackdriver.h

```diff
@@ -219,6 +222,9 @@ class StackdriverContext : public Context {
   }
   // Called on onWrite call, so counting the data that is sent.
   FilterStatus onUpstreamData(size_t size, bool) override {
+    if (!downstream_first_.has_value() && is_tcp_ && size > 0) {
+      downstream_first_ = false;
```

Comment:
Just so I am clear, this is the server writing the first byte like MySQL.
Also I would like to confirm that we only need to do this for non-tls connections, since tls is a normal client first protocol.

Reply:
I am not sure if I understand your concern. 
Assume your imagined scenario is that the downstream connection is *plaintext* protocol, and upstream is *tls*  protocol. Since it's the proxy that promote the tls(through transport socket), the arg of onUpstreamData() is only called if there is data within the tls/plaintext channel. To phrase: Tls hand shake bytes won't trigger onUpstreamData().

This PR will detect MySQL as server-first regardless the downstream/upstream are tls or plaintext